File: ios/sources/commands/ARESClosePathCommand.m
//
//  ARESClosePathCommand.m
//  ares
//
//  Created by PonyCui on 2018/11/29.
//  Copyright © 2018年 XT Studio. All rights reserved.
//

#import "ARESClosePathCommand.h"

@implementation ARESClosePathCommand

- (instancetype)init
{
    self = [super init];
    if (self) {
        [[ARESBeginPathCommand currentPath] closePath];
    }
    return self;
}

- (void)draw:(CGContextRef)ctx {
    CGContextClosePath(ctx);
}

@end

File: android/ares/src/main/java/com/xt/ares/commands/ARESClosePathCommand.kt
package com.xt.ares.commands

import android.graphics.Canvas
import com.xt.ares.ARESCommand
import com.xt.ares.ARESView

class ARESClosePathCommand: ARESCommand() {

    init {
        ARESBeginPathCommand.sharedPath.close()
    }

    override fun draw(view: ARESView, canvas: Canvas) {
        super.draw(view, canvas)
        view.currentPath.close()
    }

}